### PR TITLE
Add markdown style guide to CONTRIBUTING

### PR DESCRIPTION
**Pull request by mikekistler**
This PR adds a Markdown style guide to the CONTRIBUTING.md. I've lifted the bullets from @ralfhandl 's PRs on style fixup and added the .markdownlint.yaml file from @lornajane 's PRs for markdown updates.

There may be other things to add but I think this is a pretty good start and since we are already acting on these I thought we should make sure we have consensus and then capture that in the docs.

### CODE REVIEW

**Comment by mikekistler** [comment]
We should add the rules for "property" vs "field" vs "keyword" in #4020

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -35,6 +35,28 @@ Schemas are only changed _after_ the specification is changed.  Changes are made
 
 * Issue #3576: [What is our authoritative spec URL and how do people find it?](https://github.com/OAI/OpenAPI-Specification/issues/3576)
 
+## Style Guide
+
+Contributions to this repository should follow the style guide as described in this section.
+
+### Markdown
+
+Markdown files in this project should follow the style enforced by the [markdownlint tool][],
```
Comment (ralfhandl) [inline CONTRIBUTING.md:44]:
```suggestion
Markdown files in this project should follow the style enforced by the [markdownlint tool](https://www.npmjs.com/package/markdownlint),
```

Do we want to
```sh
npm i -D markdownlint markdownlint-cli
```
and add the updated package.json and package-lock.json to this PR?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -35,6 +35,28 @@ Schemas are only changed _after_ the specification is changed.  Changes are made
...
+as configured by the `.markdownlint.json` file in the root of the project.
+
+The following additional rules should be followed but currently are not enforced by tooling:
+
+1. The first mention of a normative reference or an OAS-defined Object in a (sub)*section is a link, additional mentions are not
+2. OAS-defined Foo Bar Objects are written in this style, and are not monospaced
+3. "example" instead of "sample" - this spec is not about statistics
+4. Use "OpenAPI Object" instead of "root"
+5. Fixed fields are monospaced
+6. Field values are monospaced in JSON notation: true, false, null, "header", ...
```
Comment (ralfhandl) [inline CONTRIBUTING.md:54]:
```suggestion
6. Field values are monospaced in JSON notation: `true`, `false`, `null`, `"header"` (with double-quotes around string values), ...
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -35,6 +35,28 @@ Schemas are only changed _after_ the specification is changed.  Changes are made
...
+7. A combination of fixed field name with example value uses JS notation: in: "header", combining rules 5 and 6
```
Comment (ralfhandl) [inline CONTRIBUTING.md:55]:
```suggestion
7. A combination of fixed field name with example value uses JS notation: `in: "header"`, combining rules 5 and 6
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -35,6 +35,28 @@ Schemas are only changed _after_ the specification is changed.  Changes are made
...
+8. An exception to 5-7 is colloquial use, for example "values of type array or object" - "type" is not monospaced, so the monospaced values aren't enclosed in double quotes.
```
Comment (ralfhandl) [inline CONTRIBUTING.md:56]:
```suggestion
8. An exception to 5-7 is colloquial use, for example "values of type `array` or `object`" - "type" is not monospaced, so the monospaced values aren't enclosed in double quotes.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -35,6 +35,28 @@ Schemas are only changed _after_ the specification is changed.  Changes are made
...
+9. "attribute" is only used in the XML context and means "XML attribute"
+10. Use Oxford commas, avoid Shatner commas
+
```
Comment (ralfhandl) [inline CONTRIBUTING.md:59]:
```suggestion

### Use of "keyword", "field", "property", and "attribute"

* JSON Schema keywords -> "keyword"
* OpenAPI fixed fields -> "field"
* property of a "plain" JSON object that is not an OpenAPI-defined Foo Object -> "property"
* "attribute" is only used in the XML context and means "XML attribute"

```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -35,6 +35,28 @@ Schemas are only changed _after_ the specification is changed.  Changes are made
...
+1. The first mention of a normative reference or an OAS-defined Object in a (sub)*section is a link, additional mentions are not
+2. OAS-defined Foo Bar Objects are written in this style, and are not monospaced
+3. "example" instead of "sample" - this spec is not about statistics
+4. Use "OpenAPI Object" instead of "root"
+5. Fixed fields are monospaced
+6. Field values are monospaced in JSON notation: true, false, null, "header", ...
+7. A combination of fixed field name with example value uses JS notation: in: "header", combining rules 5 and 6
+8. An exception to 5-7 is colloquial use, for example "values of type array or object" - "type" is not monospaced, so the monospaced values aren't enclosed in double quotes.
+9. "attribute" is only used in the XML context and means "XML attribute"
+10. Use Oxford commas, avoid Shatner commas
```
Comment (ralfhandl) [inline CONTRIBUTING.md:58]:
```suggestion
9. Use Oxford commas, avoid Shatner commas
```

**Review by ralfhandl — COMMENTED** [review]
+1, some suggestions inline

File: CONTRIBUTING.md (hunk 3 above)
Comment (mikekistler) [reply CONTRIBUTING.md:55]: Looks like I copy pasted from the rendered text and not the code, so I missed the markup. Thanks for catching this.

File: CONTRIBUTING.md (hunk 6 above)
Comment (ralfhandl) [reply CONTRIBUTING.md:58]: The "attribute" rule is now in the next subsection and can be removed here.

Comment (mikekistler) [reply CONTRIBUTING.md:58]: Sorry ... just missed this one.

**Review by ralfhandl — APPROVED** [review]

**Comment by lornajane** [comment]
In the TDC meeting we discussion that we should be using `<span id="thing"></span>` instead of the `<a name="thing"></a>` format for link anchors - it's been deprecated.

**Review by ralfhandl — APPROVED** [review]

**Comment by mikekistler** [comment]
Can I merge an "editorial" PR with just one approval?

**Comment by ralfhandl** [comment]
Unfortunately we haven't yet resolved
* #3581

**Comment by ralfhandl** [comment]
Btw., I forgot that you want to add .markdownlint.yaml in this PR in the project root.

In PRs #4040, #4041, and #4025 I add it in the `versions` folder, I can clean that up once this is merged.

Would you mind to update this PR and
```sh
npm i -D markdownlint-cli
```
so that we have the tool to interpret .markdownlint.yaml as a dev dependency?

**Review by ralfhandl — DISMISSED** [review]

**Review by ralfhandl — APPROVED** [review]

**Comment by ralfhandl** [comment]
Resolved conflict in package.json, refreshed package-lock.json

**Review by ralfhandl — APPROVED** [review]

**Comment by handrews** [comment]
@mikekistler I was intending to actually look over this by Thursday, but I have no objection to you merging it as we can always tweak it later.

**Comment by ralfhandl** [comment]
Refreshed package-lock.json due to action errors in my fork

**Review by ralfhandl — APPROVED** [review]

**Pull request merged**